apps/examples/testcase/le_tc/filesystem/itc_fs.c

```diff
@@ -0,0 +1,1442 @@
+	TC_ASSERT_EQ("close", ret, OK);
+
+	fd = open(VFS_FILE_PATH, O_RDONLY);
+	TC_ASSERT_GEQ("open", fd, 0);
+
+	ret = close(fd);
+	TC_ASSERT_EQ("close", ret, OK);
+
+	memset(read_buf, 0, BUFFER_LEN);
+	ret = read(fd, read_buf, 10);
+	TC_ASSERT_EQ("read", ret, ERROR);
+
+	ret = close(fd);
+	TC_ASSERT_EQ("close", ret, ERROR);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_fs_vfs_read_p_empty_file
+* @brief            Read file
+* @scenario         Read an empty content file
+* @apicovered       read
+* @precondition     NA
+* @postcondition    NA
+*/
+static void itc_fs_vfs_read_p_empty_file(void)
+{
+	int fd;
+	int ret;
+	char read_buf[BUFFER_LEN];
+
+	fd = open(VFS_FILE_PATH, O_WRONLY | O_TRUNC | O_CREAT);
+	TC_ASSERT_GEQ("open", fd, 0);
+	ret = close(fd);
+	TC_ASSERT_EQ("close", ret, OK);
+
+	fd = open(VFS_FILE_PATH, O_RDONLY );
+	TC_ASSERT_GEQ("open", fd, 0);
+
+	memset(read_buf, 0, BUFFER_LEN);
+	ret = read(fd, read_buf, BUFFER_LEN);
+	TC_ASSERT_EQ_CLEANUP("read", ret, 0, close(fd));
+	ret = close(fd);
+	TC_ASSERT_EQ("close", ret, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_fs_vfs_write_read_p
+* @brief            Write file, Read file
+* @scenario         Write text to the file, and read it.
+* @apicovered       write, read
+* @precondition     NA
+* @postcondition    NA
+*/
+static void itc_fs_vfs_write_read_p(void)
+{
+	int fd;
+	int ret;
+	char *write_buf = VFS_TEST_CONTENTS_1;
+	char read_buf[BUFFER_LEN];
+
+	fd = open(VFS_FILE_PATH, O_WRONLY | O_TRUNC | O_CREAT);
+	TC_ASSERT_GEQ("open", fd, 0);
+
+	ret = write(fd, write_buf, 0);
+	TC_ASSERT_EQ_CLEANUP("write", ret, 0, close(fd));
+
+	ret = write(fd, write_buf, 5);
+	TC_ASSERT_EQ_CLEANUP("write", ret, 5, close(fd));
+
+	ret = write(fd, write_buf + 5, 10);
+	TC_ASSERT_EQ_CLEANUP("write", ret, 10, close(fd));
+
+	ret = close(fd);
+	TC_ASSERT_EQ("close", ret, OK);
+
+	fd = open(VFS_FILE_PATH, O_RDONLY );
+	TC_ASSERT_GEQ("open", fd, 0);
+
+	memset(read_buf, 0, BUFFER_LEN);
+	ret = read(fd, read_buf, 0);
+	TC_ASSERT_EQ_CLEANUP("read", ret, 0, close(fd));
+
+	memset(read_buf, 0, BUFFER_LEN);
+	ret = read(fd, read_buf, 10);
+	TC_ASSERT_EQ_CLEANUP("read", ret, 10, close(fd));
+	TC_ASSERT_EQ_CLEANUP("read", strncmp(read_buf, write_buf, 10), 0, close(fd));
+
+	memset(read_buf, 0, BUFFER_LEN);
+	ret = read(fd, read_buf, 10);
+	TC_ASSERT_EQ_CLEANUP("read", ret, 5, close(fd));
+	TC_ASSERT_EQ_CLEANUP("read", strncmp(read_buf, write_buf + 10, 5), 0, close(fd));
+
+	ret = close(fd);
+	TC_ASSERT_EQ("close", ret, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_fs_vfs_fsync_p
+* @brief            Syncs the file in buffer with the disk.
+* @scenario         Open same file with two file descriptors in read and write mode, and checks fsync.
+* @apicovered       fsync, write, read
+* @precondition     NA
+* @postcondition    NA
+*/
+static void itc_fs_vfs_fsync_p(void)
+{
+	int fd;
+	int fd1;
+	int ret;
+	int len;
+	char *write_buf = VFS_TEST_CONTENTS_1;

```

Comment:
you write once, you don't have to use 2 buffers.
char buf[BUFFER_LEN];

int len = strlen(VFS_TEST_CONTENTS_1);
write(fd, buf, len);
.
.
.
memset(buf)
and then read.

Reply:
please check other place too.